pkg/binary/envoy/runtime.go

```diff
@@ -87,28 +91,37 @@ type Runtime struct {
 	RootDir  string
 	debugDir string
 	TmplDir  string
-	Config   *Config
 
 	WorkingDir string
 	IO         ioutil.StdStreams
+	Config     *Config // TODO: only used by istio until #178 deletes it
 
-	cmd           *exec.Cmd
-	fakeInterrupt context.CancelFunc
+	cmd                            *exec.Cmd
+	adminAddress, adminAddressPath string
+	fakeInterrupt                  context.CancelFunc
 
-	preStart        []func(binary.Runner) error
-	preTermination  []func(binary.Runner) error
-	postTermination []func(binary.Runner) error
+	preStart, preTermination, postTermination []func(binary.Runner) error
 
-	getAdminAddress func(r *Runtime) string
-	isReady         bool
+	isReady bool
 }
 
 // GetAdminAddress returns the current admin address in host:port format, or empty if not yet available.
+// Exported for debug.EnableEnvoyAdminDataCollection, which is always on due to debug.EnableAll.
 func (r *Runtime) GetAdminAddress() string {
```

Comment:
rather than just logging the errors here in case of failure, I would prefer having `(string, error)` return types and just return errors `return "", someError`.  Handling `GetAdminAddress() == ""` in callers looks a little bit less-Go-ish to me.

Reply:
sure I can change the signature, np!